cardano-db/src/Cardano/Db/Insert.hs

```diff
@@ -326,7 +331,7 @@ insertManyUncheckedUnique vtype records =
             . Util.commaSeparated
             $ placeholders
         , " ON CONFLICT ON CONSTRAINT "
-        , unConstraintNameDB (uniqueDBName $ onlyOneUniqueDef (Proxy @record))
+        , unConstraintNameDB foreignConstraintNameDBName
```

Comment:
foreign keys is something different from unique keys. DBSync no longer maintains any foreign keys.

I think you should take cases for this query, since sometimes the unique key constaint will be there but sometimes it won't. I'm not sure if there is a single query that can work for all cases, so db-sync needs to store the current state somewhere in memory.

Reply:
I couldn't see another way to get to the `Text` representation of a constraint name. I got it from [here](https://github.com/yesodweb/persistent/blob/eaf9d561a66a7b7a8fcbdf6bd0e9800fa525cc13/persistent-test/src/TreeTest.hs#L60)